tools/run_tests/run_tests.py

```diff
@@ -1140,6 +1140,13 @@ def test_specs(self):
                     'SCHEME': 'MacTests',
                     'PLATFORM': 'macos'
                 }))
+        out.append(
+            self.config.job_spec(
+                ['test/cpp/ios/run_tests.sh'],
```

Comment:
Put this in `CLanguage` rather than `ObjCLanguage`

Reply:
Fixed.

Reply:
Adding the test to C++ results results in C++ tests timing out. Talked with Muxi offline - this test will run as part of a new job in kokoro to avoid timeout issues.

Reply:
@muxi I have added a new test configuration: tools/internal_ci/macos/grpc_basictests_cpp_ios.cfg.